sklearn/feature_extraction/_dict_vectorizer.py

```diff
@@ -14,10 +14,20 @@
 from ..utils.validation import _deprecate_positional_args
 
 
+def _unwrap_dataframe(raw_documents):
+    # accept an iterable, or a (column-vector) dataframe
+    if hasattr(raw_documents, 'iterrows'):
+        raw_documents = raw_documents.values
+    return raw_documents
```

Comment:
I feel like our `DataFrame` handling is increasingly being scattered around. Should we just move everything to `utils._pandas` and have a better idea of how dataframes are handled?

I also find it a bit risky to check for `iterrows` and then reading `values`. I rather have a `is_pandas_df` or something maybe?

Reply:
I suppose I'm trying to support other dataframe-like entities... `.iterrows` is used because I am not trying to unwrap a Series. Though I could check `values` and `ndim`? or just unwrap the Series anyway?

Reply:
@adrinjalali – do you have a response for @jnothman?

Reply:
I think I would rather check for `values` and `ndim` instead, if that's what we're using. Although calling `.values` has quite a few consequences if it's called on a DF which stores things not supported by numpy, e.g. NA values.

On the other hand, if we're trying to get a numpy array, why not call `np.asarray(raw_documents)` instead?